.github/workflows/build-ros2-ws-action.yml

```diff
@@ -0,0 +1,91 @@
+name: CI
+on:
+  workflow_dispatch:
+  pull_request:
+  push:
+    branches: [master, feature/fix-github-workflows]
+  workflow_run:
+    workflows: [Build and Test Nightly]
+    types:  [requested, completed, in_progress]
+    branches: [feature/fix-github-workflows]
+jobs:
+  build-and-test:
+    name: Build and Test
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2.3.4
+        with:
+          path: aruku
+          ref: feature/fix-using-grpc
+
+      - name: Checkout Kansei
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/kansei          
+          path: kansei
+
+      - name: Checkout Kansei Interfaces
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/kansei_interfaces
+          path: kansei_interfaces
+      
+      - name: Checkout Aruku Interfaces
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/aruku_interfaces
+          path: aruku_interfaces
+
+      - name: Checkout Keisan
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/keisan
+          path: keisan
+
+      - name: Checkout tachimawari
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/tachimawari
+          path: tachimawari
+
+      - name: Checkout tachimawari interfaces
+        uses: actions/checkout@v2.3.4
+        with:
+          repository: ichiro-its/tachimawari_interfaces
+          path: tachimawari_interfaces
```

Comment:
To make it simple, maybe we can just git clone them all like what we did in the setup abseil and grpc steps. I don't know though what is the real benefit of using the `actions/checkout` compared to the `git clone` command for cloning a repository other than the current repository.

Reply:
Maybe `actions/checkout` will help with the ethics of wanting to clone from a private repository because it provides parameters to add an SSH key mas.

Reply:
which private repository we're going to clone in this case?

Reply:
Oh, there isn't any in this repository, but does this also change all actions in all repositories? From those that previously used checkout to regular git clone?

Reply:
> which private repository we're going to clone in this case?

Oh, there isn't any in this repository, but does this also change all actions in all repositories? From those that previously used checkout to regular git clone?

Reply:
It's actually up to you, and I also don't know what's best for this case.

But here's some insight:
- It seems that what makes [`actions/checkout`](https://github.com/actions/checkout) commonly used in GitHub workflows is that it handles checking out the current repository on a specific commit depending on the event (hence the name Checkout action). In this case, you may not need to use `actions/checkout` if the objective is to clone external repositories.
- In this action, we need to clone multiple repositories. In GitHub workflows, people normally use `actions/checkout` to clone repositories (including external repositories). However, if we use this, we'll have a lot of steps just for cloning the dependencies, which in my opinion looks messy.
- As an alternative, maybe we can run the step as a script and call multiple `git clone` commands in the same step. Remember that you don't need [a sledgehammer to crack a nut](https://dictionary.cambridge.org/dictionary/english/sledgehammer-to-crack-a-nut).
- There's also another alternative like [`vweevers/multi-checkout-action`](https://github.com/vweevers/multi-checkout-action) which supports checking out multiple repos in the same step and even supports private repositories.

In short, my point is not about the `git clone` command itself, but having to specify multiple steps just for cloning external dependencies seems messy to me. However, it's also fine if you prefer to keep this style.

Reply:
oh i see, okay mas I'll try to simplify everything by just using 'git clone' for the external repository.